### ISSUE TITLE
Can't point to a specific version for pip

### ISSUE THREAD

**Issue by rtpg** [xonsh/xonsh#1754]
Writing this here for posterity, Hopefully I'll be able to find the root cause this weekend.

Basically I can't run pip properly. I'm assuming this is a parsing issue related to mode detection, but the first part of the command is very "command-y" so I'm a bit surprised.

```
>>> pip install django==1.9.9
...
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 16237, in _parse_ctx_free
    debug_level=(self.debug_level > 1))
  File "/usr/local/lib/python3.5/site-packages/xonsh/parsers/base.py", line 314, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "/usr/local/lib/python3.5/site-packages/ply/yacc.py", line 331, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/usr/local/lib/python3.5/site-packages/ply/yacc.py", line 1181, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "/usr/local/lib/python3.5/site-packages/ply/yacc.py", line 193, in call_errorfunc
    r = errorfunc(token)
  File "/usr/local/lib/python3.5/site-packages/xonsh/parsers/base.py", line 2355, in p_error
    self.xonsh_code)
  File "/usr/local/lib/python3.5/site-packages/xonsh/parsers/base.py", line 420, in _parse_error
    raise err
  File "<string>", line None
SyntaxError: <xonsh-code>:1:20: ('code: ==',)
![pip install django==1.9.9]
                    ^

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/xonsh/ptk/shell.py", line 99, in push
    locs=None)
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 16184, in compile
    tree = self.parse(input, ctx, mode=mode, transform=transform)
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 16155, in parse
    tree, input = self._parse_ctx_free(input, mode=mode)
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 16295, in _parse_ctx_free
    raise original_error
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 16237, in _parse_ctx_free
    debug_level=(self.debug_level > 1))
  File "/usr/local/lib/python3.5/site-packages/xonsh/parsers/base.py", line 314, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "/usr/local/lib/python3.5/site-packages/ply/yacc.py", line 331, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/usr/local/lib/python3.5/site-packages/ply/yacc.py", line 1181, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "/usr/local/lib/python3.5/site-packages/ply/yacc.py", line 193, in call_errorfunc
    r = errorfunc(token)
  File "/usr/local/lib/python3.5/site-packages/xonsh/parsers/base.py", line 2355, in p_error
    self.xonsh_code)
  File "/usr/local/lib/python3.5/site-packages/xonsh/parsers/base.py", line 420, in _parse_error
    raise err
  File "<string>", line None
SyntaxError: <xonsh-code>:1:4: ('code: install',)
pip install django==1.9.9
    ^
```

**Comment by laerus** [comment]
Hey @rtpg thanks for reporting! I think this falls in the #1432 category where you can quote the part that is causing the issue like so `pip install 'django==1.9.9'`

**Comment by scopatz** [comment]
Thanks for reporting @rtpg! Would you mind trying out the fix in #1759.  Otherwise @laerus's work around with strings should work too
